nextcord/application_command.py

```diff
@@ -0,0 +1,883 @@
+    pass
+
+
+class ApplicationCommandResponse(Hashable):
+    """Represents the response that Discord sends back when queried for Application Commands.
+
+    Attributes
+    ----------
+    id: :class:`int`
+        Discord ID of the Application Command.
+    type: :class:`nextcord.ApplicationCommandType`
+        Enum corresponding to the Application Command type. (slash, message, user)
+    guild_id: Optional[:class:`int`]
+        The Guild ID associated with the Application Command. If None, it's a global command.
+    name: :class:`str`
+        Name of the Application Command.
+    description: :class:`str`
+        Description of the Application Command.
+    options: List[:class:`nextcord.ApplicationCommandResponseOption`]
+        A list of options or subcommands that the Application Command has.
+    default_permission: :class:`bool`
+        If the command is enabled for users by default.
+    """
+
+    def __init__(self, state: ConnectionState, payload: dict):
+        self._state: ConnectionState = state
+        self.id: int = int(payload["id"])
+        self.type: ApplicationCommandType = ApplicationCommandType(payload["type"])
+        self.guild_id: Optional[int] = utils._get_as_snowflake(payload, "guild_id")
+        self.name: str = payload["name"]
+        self.description: str = payload["description"]
+        self.options: List[
+            ApplicationCommandResponseOption
+        ] = ApplicationCommandResponseOption._create_options(payload.get("options", []))
+        self.default_permission: Optional[bool] = payload.get(
+            "default_permission", True
+        )
+
+    @property
+    def guild(self) -> Optional[Guild]:
+        """Optional[:class:`Guild`]: Returns the :class:`Guild` associated with this Response, if any."""
+        return self._state._get_guild(self.guild_id)
+
+    @property
+    def signature(self) -> Tuple[str, int, Optional[int]]:
+        """Returns a simple high level signature of the command. No commands registered in the bot at the same time
+        should have identical signatures.
+
+        Returns
+        -------
+        name: :class:`str`
+            Name of the Application Command.
+        type: :class:`int`
+            Discord's integer value of the Application Command type
+        guild_id: Optional[:class:`int`]
+            The Guild ID associated with the Application Command. If None, it's a global command.
+        """
+        return self.name, self.type.value, self.guild_id
+
+
+class ApplicationCommandResponseOptionChoice:
+    """Represents a single choice in a list of options.
+
+    Attributes
+    ----------
+    name: :class:`str`
+        Name of the choice, this is what users see in Discord.
+    value: Union[:class:`str`, :class:`int`, :class:`float`]
+        Value of the choice, this is what Discord sends back to us.
+    """
+
+    def __init__(self, payload: Optional[dict] = None):
```

Comment:
Given we support 3.8, you need to use ``typing.Dict`` here. Directly typing standard collections came in 3.9 iirc

Reply:
Not sure. I'll test today, but it'd be consistent if it's changed given the parts when typing the dict already using the typing one

Reply:
Yea looks fine, guess its just a consistency thing